src/python/deepgnn/graph_engine/snark/tests/convert_test.py

```diff
@@ -722,6 +776,378 @@ def test_sanity_edge_sparse_features_data(graph_with_sparse_features):
+        npt.assert_almost_equal(
+            np.frombuffer(result[80:88], dtype=np.float32), [5.5, 6.6]
+        )
+        npt.assert_equal(np.frombuffer(result[88:96], dtype=np.int32), [5, 6])
+
+
+def test_edge_list_header_multiple_partitions():
+    output = tempfile.TemporaryDirectory()
+    data_data = [
+        "0,-1\n",
+        "1,-1\n",
+        "2,-1\n",
+    ]
+    meta_data = {"default_node_type": 0, "default_node_weight": 1.5}
+    _gen_edge_list(output, data_data, meta_data, partitions=2)
+    with open("{}/node_{}_{}.map".format(output.name, 0, 0), "rb") as nm:
+        expected_size = 2 * (2 * 8 + 4)
+        result = nm.read(expected_size + 8)
+        assert len(result) == expected_size
+        assert result[0:8] == (0).to_bytes(8, byteorder=sys.byteorder)
+        assert result[8:16] == (0).to_bytes(8, byteorder=sys.byteorder)
+        assert result[16:20] == (0).to_bytes(4, byteorder=sys.byteorder)
+        assert result[20:28] == (2).to_bytes(8, byteorder=sys.byteorder)
+        assert result[28:36] == (1).to_bytes(8, byteorder=sys.byteorder)
+        assert result[36:40] == (0).to_bytes(4, byteorder=sys.byteorder)
+
+
+def test_edge_list_error_checking():
+    decoder = EdgeListDecoder()
+    with pytest.raises(StopIteration):
+        next(decoder.decode(""))
+    with pytest.raises(ValueError):
+        next(decoder.decode("x"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1,4"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,x"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1,4,1,bad_key"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1,4,1,float32"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1,x"))
+    next(decoder.decode("0,-1,4,1,float32,2,1,1"))
+    next(decoder.decode("0,-1,4,1,binary,1,test"))
+    next(decoder.decode("0,-1,4,1,float32,0"))
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1")
+        next(gen)
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1,4")
+        next(gen)
+    gen = decoder.decode("0,1,4,1")
+    next(gen)
+
+
+def test_edge_list_binary_escape():
+    decoder = EdgeListDecoder()
+    src, dst, typ, weight, features = next(decoder.decode("0,-1,0,1.0,binary,1,test"))
+    assert features[0] == "test"
+    with pytest.raises(RuntimeError):
+        src, dst, typ, weight, features = next(
+            decoder.decode("0,-1,0,1.0,binary,1,test,feature")
+        )
+    src, dst, typ, weight, features = next(
+        decoder.decode("0,-1,0,1.0,binary,1,test\,feature")
```

Comment:
Please, make sure that it is ok to use spaces after comas:
```python
    src, dst, typ, weight, features = next(
        decoder.decode("0, -1, 0, 1.0, binary, 1,test\,feature")
    )
```

Reply:
Users can enable this by overriding the delimiter=", " which is tested.

Reply:
", " is not a proper delimiter, because parsers typically skip spaces.